char/char.mbt

```diff
@@ -19,3 +19,300 @@ pub impl Hash for Char with hash(self) { self.to_int() }
+/// - U+0061 ‘a’ ..= U+007A ‘z’
+pub fn is_ascii_alphabetic(self : Char) -> Bool {
+  match self {
+    'A'..='Z' | 'a'..='z' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII control character:
+/// U+0000 NUL ..= U+001F UNIT SEPARATOR, or U+007F DELETE.
+/// Note that most ASCII whitespace characters are control characters, but SPACE is not.
+pub fn is_ascii_control(self : Char) -> Bool {
+  match self {
+    '\x00'..='\x1F' | '\x7F' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII decimal digit:
+/// U+0030 ‘0’ ..= U+0039 ‘9’
+pub fn is_ascii_digit(self : Char) -> Bool {
+  match self {
+    '0'..='9' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII graphic character:
+/// U+0021 ‘!’ ..= U+007E ‘~’
+pub fn is_ascii_graphic(self : Char) -> Bool {
+  match self {
+    '\x21'..='\x7E' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII hexadecimal digit:
+/// - U+0030 ‘0’ ..= U+0039 ‘9’
+/// - U+0041 ‘A’ ..= U+0046 ‘F’
+/// - U+0061 ‘a’ ..= U+0066 ‘f’
+pub fn is_ascii_hexdigit(self : Char) -> Bool {
+  match self {
+    '0'..='9' | 'A'..='F' | 'a'..='f' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII lowercase character:
+/// U+0061 ‘a’ ..= U+007A ‘z’.
+pub fn is_ascii_lowercase(self : Char) -> Bool {
+  match self {
+    'a'..='z' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII octal digit:
+/// U+0030 ‘0’ ..= U+0037 ‘7’
+pub fn is_ascii_octdigit(self : Char) -> Bool {
+  match self {
+    '0'..='7' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII punctuation character:
+/// - U+0021 ..= U+002F ! " # $ % & ' ( ) * + , - . /
+/// - U+003A ..= U+0040 : ; < = > ? @
+/// - U+005B ..= U+0060 [ \ ] ^ _ `
+/// - U+007B ..= U+007E { | } ~
+pub fn is_ascii_punctuation(self : Char) -> Bool {
+  match self {
+    '\x21'..='\x2F' | '\x3A'..='\x40' | '\x5B'..='\x60' | '\x7B'..='\x7E' =>
+      true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII uppercase character:
+/// U+0041 ‘A’ ..= U+005A ‘Z’
+pub fn is_ascii_uppercase(self : Char) -> Bool {
+  match self {
+    'A'..='Z' => true
+    _ => false
+  }
+}
+
+///| Checks if the value is an ASCII whitespace character:
+/// U+0020 SPACE, U+0009 HORIZONTAL TAB, U+000A LINE FEED, U+000C FORM FEED, or U+000D CARRIAGE RETURN.
+pub fn is_ascii_whitespace(self : Char) -> Bool {
+  match self {
+    '\x20' | '\x09' | '\x0A' | '\x0C' | '\x0D' => true
+    _ => false
+  }
+}
+
+///| Returns true if this char has the general category for control codes.
+pub fn is_control(self : Char) -> Bool {
+  match self {
+    '\u0000'..='\u001F' | '\u007F'..='\u009F' => true
+    _ => false
+  }
+}
+
+///| Checks if a char is a digit in the given radix (range from 2 to 36).
+pub fn is_digit(self : Char, radix : UInt) -> Bool!Error {
+  guard radix >= 2 && radix <= 36 else {
+    fail!("radix must be in the range from 2 to 36")
```

Comment:
This should panic instead of failing; the error is rare and should be considered user error. Will fix in another PR.